File: C_programs/src/Program_5.c
#include <stdlib.h>
#include <Program_5.h>

int* concatenateArrayTwice(int arr[], const int size) {
    // Create a new array of double the size
    const int newSize = size * 2;
    int* newArr = malloc(newSize * sizeof(int));

    // Copy elements of the original array into the new array twice
    for (int i = 0; i < size; i++) {
        newArr[i] = arr[i];
        newArr[i + size] = arr[i];
    }

    return newArr;
}

File: Python_Programs/src/Program_5.py
def concatenate_array_twice(arr):
    return arr * 2

"""
def concatenate_array_twice(arr):
    # Create a new list with double the size
    new_size = len(arr) * 2
    new_arr = [0] * new_size

    # Copy elements of the original list into the new list twice
    for i in range(len(arr)):
        new_arr[i] = arr[i]
        new_arr[i + len(arr)] = arr[i]

    return new_arr
"""
